acados/sim/sim_gnsf.c

```diff
@@ -0,0 +1,1565 @@
+{
+    int size = sizeof(sim_gnsf_dims);
+    return size;
+}
+
+
+void *sim_gnsf_dims_assign(void* config_, void *raw_memory)
+{
+    char *c_ptr = (char *) raw_memory;
+    sim_gnsf_dims *dims = (sim_gnsf_dims *) c_ptr;
+    c_ptr += sizeof(sim_gnsf_dims);
+    assert((char *) raw_memory + sim_gnsf_dims_calculate_size() == c_ptr);
+    return dims;
+}
+
+/************************************************
+* get & set functions
+************************************************/
+
+void sim_gnsf_set_nx(void *dims_, int nx)
+{
+    sim_gnsf_dims *dims = (sim_gnsf_dims *) dims_;
+    dims->nx = nx;
+}
+
+void sim_gnsf_set_nu(void *dims_, int nu)
+{
+    sim_gnsf_dims *dims = (sim_gnsf_dims *) dims_;
+    dims->nu = nu;
+}
+
+void sim_gnsf_get_nx(void *dims_, int* nx)
+{
+    sim_gnsf_dims *dims = (sim_gnsf_dims *) dims_;
+    *nx = dims->nx;
+}
+
+void sim_gnsf_get_nu(void *dims_, int* nu)
+{
+    sim_gnsf_dims *dims = (sim_gnsf_dims *) dims_;
+    *nu = dims->nu;
+}
+
+/************************************************
+* import functions
+************************************************/
+
+void gnsf_import_matrices(sim_gnsf_dims* dims, gnsf_model *model, external_function_generic *get_matrices_fun)
+{
+    // calling the external function
+    ext_fun_arg_t ext_fun_type_in[1];
+	void *ext_fun_in[1];
+    ext_fun_arg_t ext_fun_type_out[9];
+	void *ext_fun_out[9];
+
+    ext_fun_type_in[0] = COLMAJ;
+    ext_fun_in[0] = model->A_dt; // just to have some input;
+
+    ext_fun_type_out[0] = COLMAJ;
+    ext_fun_type_out[1] = COLMAJ;
+    ext_fun_type_out[2] = COLMAJ;
+    ext_fun_type_out[3] = COLMAJ;
+    ext_fun_type_out[4] = COLMAJ;
+    ext_fun_type_out[5] = COLMAJ;
+    ext_fun_type_out[6] = COLMAJ;
+    ext_fun_type_out[7] = COLMAJ;
+    ext_fun_type_out[8] = COLMAJ;
+
+    ext_fun_out[0] = model->A;
+    ext_fun_out[1] = model->B;
+    ext_fun_out[2] = model->C;
+    ext_fun_out[3] = model->E;
+    ext_fun_out[4] = model->L_x;
+    ext_fun_out[5] = model->L_xdot;
+    ext_fun_out[6] = model->L_z;
+    ext_fun_out[7] = model->L_u;
+    ext_fun_out[8] = model->A_LO;
+
+
+    get_matrices_fun->evaluate(get_matrices_fun, ext_fun_type_in, ext_fun_in, ext_fun_type_out, ext_fun_out);
+}
+
+/************************************************
+* OUTDATED FUNCTIONS: the following functions were implemented
+* before some major changes were applied, maybe similar
+* functionalities will be implemented at some point:
+************************************************/
+// void gnsf_get_dims( sim_gnsf_dims *dims, casadi_function_t get_ints_fun)
+
+// void gnsf_import_precomputed(sim_gnsf_dims* dims, gnsf_model *model, casadi_function_t But_KK_YY_ZZ_LO_fun)
+// this function can be used instead of import_matrices + precompute (not working now, ATTENTION!!!)
+
+
+
+/************************************************
+* helpful functions
+************************************************/
+
+double minimum_of_doubles(double *x, int n){
```

Comment:
this stuff doesn't really belong here... since it is used only in the examples, I would move it there

Reply:
I talked with @dkouzoup and moved it to `utils/math.c`.